loginLegacy/src/main/java/com/brandon/loginlegacy/view/LoginScreenActivity.kt

```diff
@@ -25,64 +27,72 @@ class LoginScreenActivity : AppCompatActivity() {
         binding = LoginScreenBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        val emailField = binding.loginEmailField
-        val emailError = binding.loginEmailError
-        val codeField = binding.loginCodeField
-        val fieldsError = binding.loginRequiredFieldsError
-        val buttonSpinner = binding.progressSpinner
-        val buttonText = binding.buttonText
-        val loginButton = binding.loginButton
+        loginViewModel.viewModelScope.launch {
+            loginViewModel.uiState.collect {
+                updateUi(it)
+                Log.e(LoginViewModel.VIEWMODEL_ERROR_TAG, "${loginViewModel.uiState}")
+            }
+        }
 
         binding.loginButton.setOnClickListener {
-            fieldsError.visibility = GONE
-            val email = binding.loginEmailField.text.toString()
-            val code = binding.loginCodeField.text.toString()
+            val emailField = binding.loginEmailField
+            val codeField = binding.loginCodeField
+            val fieldsError = binding.loginRequiredFieldsError
+            fieldsError.isVisible = true
+            emailField.clearFocus()
+            emailField.isSelected = false
+            val email = emailField.text.toString()
+            val code = codeField.text.toString()
             if (email.isNotEmpty()) {
                 if (code.isNotEmpty()) loginViewModel.loginClicked(email, code)
                 else loginViewModel.loginClicked(email)
-            } else fieldsError.visibility = VISIBLE
+            } else fieldsError.isVisible = true
         }
 
         binding.createAccountRedirect.setOnClickListener {
             // TODO FA-101 Navigate to Create Account Page
         }
+    }
 
-        loginViewModel.onUpdate = {
-            when (it) {
+    private fun updateUi(uiState: LoginViewModel.LoginUiState) {
+        binding.apply {
+            when (uiState) {
                 is Idle -> {
-                    buttonSpinner.visibility = GONE
-                    buttonText.visibility = VISIBLE
-                    codeField.visibility = if (it.isCodeSent) VISIBLE else GONE
-                    emailError.visibility = if (it.isEmailError) VISIBLE else GONE
-
+                    progressSpinner.isVisible = false
+                    buttonText.isVisible = true
+                    loginCodeField.apply {
+                        isVisible = uiState.isCodeSent
+                        isEnabled = true
+                    }
+                    loginEmailError.isVisible = uiState.isEmailError
                     loginButton.isEnabled = true
-                    emailField.enable(InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS)
-                    codeField.enable(InputType.TYPE_CLASS_NUMBER)
-
-                    emailField.text = it.email.toEditable()
+                    loginEmailField.apply {
+                        isEnabled = true
+                        text = uiState.email.toEditable()
```

Comment:
I believe you can just use `setText` and not need the `toEditable`

Reply:
uses setText now